google_guest_agent/hostname/hostname_windows.go

```diff
@@ -0,0 +1,120 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package hostname
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+	"syscall"
+	"unsafe"
+
+	"github.com/GoogleCloudPlatform/guest-agent/google_guest_agent/command"
+	"github.com/GoogleCloudPlatform/guest-agent/utils"
+	"github.com/GoogleCloudPlatform/guest-logging-go/logger"
+	"golang.org/x/sys/windows"
+)
+
+const (
+	platformHostsFile = `C:\Windows\System32\Drivers\etc\hosts`
+	newline           = "\r\n"
+)
+
+var (
+	ipcallbackHandleMu          = new(sync.Mutex)
+	ipcallbackHandle            uintptr
+	iphlpapi                    = windows.NewLazySystemDLL("iphlpapi.dll")
+	procNotifyIpInterfaceChange = iphlpapi.NewProc("NotifyIpInterfaceChange")
+	procCancelMibChangeNotify2  = iphlpapi.NewProc("CancelMibChangeNotify2")
+)
+
+var setHostname = func(hostname string) error {
+	return fmt.Errorf("setting hostnames in guest-agent is not supported on windows")
+}
+
+func notifyIpInterfaceChange(family uint32, callbackPtr uintptr, callerContext unsafe.Pointer, initialNotif bool, handle *uintptr) error {
+	notify := 0
+	if initialNotif {
+		notify = 1
+	}
+
+	r, _, e := procNotifyIpInterfaceChange.Call(
```

Comment:
why not use `syscall`?

Reply:
That's the convention in existing agent code (`accounts_windows.go`). Do you want to start switching to `SyscallN`?

Reply:
Done.